app/code/Magento/InventoryInStorePickupApi/Exception/OrderIsNotReadyForPickupException.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupApi\Exception;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Phrase;
+
+/**
+ * Class OrderIsNotReadyForPickupException
+ *
+ * @package Magento\InventoryInStorePickupApi\Exception
+ * @api
+ */
+class OrderIsNotReadyForPickupException extends LocalizedException
```

Comment:
What is the reason for having one more exception type?

Reply:
Because we can have `LocalizedException` which means that something went wrong in general during the processing, but I thought it would be useful to be able to check this specific case (like NoSuchEntityException).
Anyway, there is no problem to remove it.

Reply:
This would encourage building business logic using exceptions instead of conditional logic. Existing exceptions serve additional purpose of resolving HTTP codes, like NoSuchEntity indicates 404.

Reply:
`OrderIsNotReadyForPickupException` have been killed.